components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilderTest.java

```diff
@@ -0,0 +1,174 @@
+
+import java.net.SocketException;
+
+import static org.mockito.Mockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertNotNull;
+
+@PrepareForTest({ServerConfiguration.class, CarbonUtils.class, IdentityCoreServiceComponent.class, NetworkUtils.class,
+        IdentityTenantUtil.class, URLResolverService.class, PrivilegedCarbonContext.class})
+@PowerMockIgnore({"javax.net.*", "javax.security.*", "javax.crypto.*", "javax.xml.*", "org.xml.sax.*", "org.w3c.dom" +
+        ".*", "org.apache.xerces.*"})
+public class DefaultServiceURLBuilderTest {
+
+    @Mock
+    private ServerConfiguration mockServerConfiguration;
+    @Mock
+    private ConfigurationContextService mockConfigurationContextService;
+    @Mock
+    private ConfigurationContext mockConfigurationContext;
+    @Mock
+    private AxisConfiguration mockAxisConfiguration;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        mockStatic(CarbonUtils.class);
+        mockStatic(ServerConfiguration.class);
+        mockStatic(NetworkUtils.class);
+        mockStatic(IdentityCoreServiceComponent.class);
+        mockStatic(IdentityTenantUtil.class);
+        mockStatic(PrivilegedCarbonContext.class);
+        PrivilegedCarbonContext privilegedCarbonContext = mock(PrivilegedCarbonContext.class);
+
+        when(PrivilegedCarbonContext.getThreadLocalCarbonContext()).thenReturn(privilegedCarbonContext);
+        when(ServerConfiguration.getInstance()).thenReturn(mockServerConfiguration);
+        when(IdentityCoreServiceComponent.getConfigurationContextService()).thenReturn(mockConfigurationContextService);
+        when(mockConfigurationContextService.getServerConfigContext()).thenReturn(mockConfigurationContext);
+        when(mockConfigurationContext.getAxisConfiguration()).thenReturn(mockAxisConfiguration);
+        try {
+            when(NetworkUtils.getLocalHostname()).thenReturn("localhost");
+        } catch (SocketException e) {
+            // Mock behaviour, hence ignored
+        }
+
+        System.setProperty(IdentityConstants.CarbonPlaceholders.CARBON_PORT_HTTP_PROPERTY, "9763");
+        System.setProperty(IdentityConstants.CarbonPlaceholders.CARBON_PORT_HTTPS_PROPERTY, "9443");
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        System.clearProperty(IdentityConstants.CarbonPlaceholders.CARBON_PORT_HTTP_PROPERTY);
+        System.clearProperty(IdentityConstants.CarbonPlaceholders.CARBON_PORT_HTTPS_PROPERTY);
+    }
+
+    @Test
+    public void testAddPath() {
+
+        String testPath = "/testPath";
+        ServiceURL serviceURL = null;
+        try {
+            serviceURL = ServiceURLBuilder.create().addPath(testPath).build();
+        } catch (URLBuilderException e) {
+            // Mock behaviour, hence ignored
+        }
+
+        assertNotNull(serviceURL.getUrlPath());
```

Comment:
In each of these methods we should check if the URL is built as expected. Null check is not enough

Reply:
basically verify for absolute url, relative url and the respective component like fragment, parameter etc.

Reply:
That is done in the ServiceURLTest class.

ref: https://github.com/wso2/carbon-identity-framework/blob/4d5669512d85c459c670d72dfe0692fa7dddeeda/components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/model/ServiceURLTest.java#L182

Reply:
Then why check null. Shouldn't the expectation is to verify the path added